project-icp/bee-pollinator-app
# Monthly Survey Form

## Overview

Adds a Monthly Survey form for taking in monthly surveys. Each form has a main `num_colonies` textbox, and then three tabs with the detailed forms.

The back-end endpoints were updated to allow sending multiple entries. Existing survey forms were updated accordingly.

Only those colonies with a "name" are saved. We should add a check to ensure there is at least one colony in the survey before saving it.

**TODO**

- [x] Add a check so that they have to fill out at least one colony
- [x] Add text to show that if the colony name is not entered, it will not be counted
- [x] Rebase on top of #416 to incorporate success modal
- [x] Make submission code match with April and November
- [x] monthly surveys should appear on the 15th of that month (i.e. July's survey appears on July 15)

Connects #381 

### Demo

![image](https://user-images.githubusercontent.com/1430060/51194979-27123880-18ba-11e9-8199-a3688c31bd4e.png)

### Notes

Lacks styling and validation, as do the other forms.

## Testing Instructions

* Check out this branch
* Run migrations `./scripts/manage.sh migrate`
* Start the app `./scripts/beekeepers.sh start`
* Log in to a "Pro" user account
* Go to surveys. Enter in Monthly surveys. Ensure they are saved correctly.
* Enter in November and April surveys. Ensure they still work.
* Load existing Monthly surveys. Ensure they are populated correctly.
